go/vt/vtgate/engine/insert.go

```diff
@@ -572,22 +583,20 @@ func (ins *Insert) processUnowned(vcursor VCursor, vindexColumnsKeys [][]sqltype
 
 	// For cases where a value was not supplied, we reverse map it
 	// from the keyspace id, if possible.
-	if reverseKsids != nil {
-		reversible, ok := colVindex.Vindex.(vindexes.Reversible)
-		if !ok {
-			return fmt.Errorf("value must be supplied for column %v", colVindex.Columns)
-		}
-		reverseKeys, err := reversible.ReverseMap(vcursor, reverseKsids)
+	if isReversible && len(reverseKsids) > 0 {
+		reverseKeys, err := reversibleVindex.ReverseMap(vcursor, reverseKsids)
 		if err != nil {
 			return err
 		}
+
 		for i, reverseKey := range reverseKeys {
 			// Fill the first column with the reverse-mapped value.
 			vindexColumnsKeys[reverseIndexes[i]][0] = reverseKey
 		}
 	}
 
-	if verifyKsids != nil {
+	// Verify keyspace ids from primary and secondary VIndexes match
+	if len(verifyIndexes) > 0 {
```

Comment:
My understanding of Go is not great, but I believe the `verifyKsids != nil` check was always returning `true`, because `verifyKsids` is declared as an empty slice, not a nil slice. Using `len(verifyIndexes) > 0` is more in line of the actual intention of this code.

Reply:
Apparently using `nil` slices is preferred in Go, according to https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices. I'll make that change and push it up shortly.